lib/model/Fixture.js

```diff
@@ -128,6 +128,32 @@ module.exports = class Fixture {
     return this._cache.allChannels;
   }
 
+  // returns object whose keys are all items in this.allChannelKeys
+  // { 'channel key': 'unique channel name' }
+  get uniqueChannelNames() {
+    if (!('uniqueChannelNames' in this._cache)) {
+      this._cache.uniqueChannelNames = {};
+
+      let names = this.allChannels.map(ch => ch.name || ch.key);
+
+      for (let i = 0; i < names.length; i++) {
+        const originalName = names[i];
+
+        // make unique by appending ' 2', ' 3', ...
+        let duplicates = 1;
+        while (names.indexOf(names[i]) !== i) {
+          duplicates++;
+          names[i] = `${originalName} ${duplicates}`;
```

Comment:
Better add another variable that you save to to make it more clear. It isn't used in the array anymore anyways.